ERCS/erc-8021.md

```diff
@@ -0,0 +1,261 @@
+
+**ASCII encoding**: Using human-readable ASCII codes improves debuggability and allows for meaningful code names rather than opaque hashes.
+
+**ERC-5792 integration**: The capability-based approach aligns with ERC-5792's extensible design while providing a clean separation of concerns between applications and wallets.
+
+### Alternative Approaches Considered
+
+**Customizable registry**: While a single global registry is recommended when available for reduced transaction data, the flexible registry approach allows different ecosystems to maintain their own attribution systems.
+
+**Fixed-length codes**: A fixed-length encoding was considered but rejected to allow for flexible, human-readable identifiers.
+
+**ERC-721 dependency**: While codes may be implemented as ERC-721 NFTs, reducing strict dependencies makes this proposal more flexible and clarifies the minimal interface truly required for attribution.
+
+**Structured ERC-5792 capability**: A generic bytes `dataSuffix` minimizes the effort on wallets to integrate and maintain support for new schemaIds, increasing chances of wide adoption.
+
+## Backwards Compatibility
+
+This standard is fully backward compatible with existing transaction formats. Transactions without attribution suffixes continue to function normally, and the presence of attribution data does not affect transaction execution.
+
+Existing applications using informal attribution methods can migrate to this standard while maintaining their current functionality during the transition period.
+
+## Security Considerations
+
+### Parsing Safety
+
+Parsers MUST validate the suffix format before processing to prevent malformed data from causing errors. Invalid suffixes SHOULD be ignored rather than causing parsing failures.
+
+### Registry Trust
+
+The security of attribution data depends on the trustworthiness of the registry contract. Users and applications SHOULD verify the registry implementation and governance before relying on its data.
+
+### Data Integrity
+
+Since attribution data is appended to calldata, it could potentially be modified by intermediaries. Applications requiring cryptographic proof of attribution should implement additional signature-based verification mechanisms.
+
+### Privacy Considerations
+
+Attribution data is publicly visible on-chain. Applications handling sensitive user information SHOULD NOT include identifying information in attribution codes.
+
+## Test Cases
+
+Single entity attribution + canonical registry:
+```
+Input: 0xdddddddd62617365617070070080218021802180218021802180218021
+Expected: 
+- txData: 0xdddddddd
+- schemaId: 0
+- codes: ["baseapp"]
+- registry: canonical
+```
+
+Multiple entity attribution + custom registry:
+```
+Input: 0xdddddddd626173656170702C6D6F7270686F0Ecccccccccccccccccccccccccccccccccccccccc0180218021802180218021802180218021
```

Comment:
In the definition there is the code before and the registry address after, here are inverted, what is the right direction?